### PR TITLE
new ExprTimePlayed

### PR DESCRIPTION
**Pull request by ShaneBeee**
### Description
Adds an expression to get the amount of time a player has played on the server for.
ex:
```vb
if player's time played is greater than 10 days:
    set {legacy::%uuid of player%} to true
```

---
**Target Minecraft Versions:** any
**Requirements:** none
**Related Issues:**  none

### CODE REVIEW

File: src/main/java/ch/njol/skript/expressions/ExprTimePlayed.java [hunk 1]
```diff
@@ -0,0 +1,118 @@
...
+	"set player's time played to 0 seconds"})
+@Since("INSERT VERSION")
+public class ExprTimePlayed extends SimplePropertyExpression<Player, Timespan> {
+	
+	private final static Statistic TIME_PLAYED;
+
+	static {
+		register(ExprTimePlayed.class, Timespan.class, "time played", "players");
+		if (Skript.isRunningMinecraft(1, 13)) {
+			TIME_PLAYED = Statistic.PLAY_ONE_MINUTE;
```
Comment (Wealthyturtle) [inline src/main/java/ch/njol/skript/expressions/ExprTimePlayed.java]: Should probably add a comment that the ``PLAY_ONE_MINUTE`` statistic is misleading and actually records the ticks online

File: src/main/java/ch/njol/skript/expressions/ExprTimePlayed.java [hunk 2]
```diff
@@ -0,0 +1,118 @@
...
+	@Override
+	public boolean init(Expression<?>[] exprs, int matchedPattern, Kleenean isDelayed, ParseResult parseResult) {
+		setExpr((Expression<Player>) exprs[0]);
+		return true;
+	}
+	
+	@Nullable
+	@Override
+	public Timespan convert(Player player) {
+		return new Timespan((long) player.getStatistic(TIME_PLAYED) * 50);
```
Comment (Blueyescat) [inline src/main/java/ch/njol/skript/expressions/ExprTimePlayed.java]:
```suggestion
		return Timespan.fromTicks_i((long) player.getStatistic(TIME_PLAYED));
```

File: src/main/java/ch/njol/skript/expressions/ExprTimePlayed.java [hunk 3]
```diff
@@ -0,0 +1,118 @@
...
+		return "time played";
+	}
+	
+	@Override
+	public Class<? extends Timespan> getReturnType() {
+		return Timespan.class;
+	}
+	
+	@Override
+	public String toString(@Nullable Event e, boolean d) {
```
Comment (Blueyescat) [inline src/main/java/ch/njol/skript/expressions/ExprTimePlayed.java]: no toString for a SimplePropExpr lol

File: src/main/java/ch/njol/skript/expressions/ExprTimePlayed.java [hunk 4]
```diff
@@ -0,0 +1,118 @@
...
+				case REMOVE:
+					ticks = playerTicks - ticks;
+					break;
+			}
+			player.setStatistic(TIME_PLAYED, (int) ticks);
+		}
+	}
+	
+	@Override
+	protected String getPropertyName() {
```
Comment (Blueyescat) [inline src/main/java/ch/njol/skript/expressions/ExprTimePlayed.java]: getPropertyName should be at the end

File: src/main/java/ch/njol/skript/expressions/ExprTimePlayed.java [hunk 5]
```diff
@@ -0,0 +1,118 @@
...
+import ch.njol.skript.doc.Since;
+import ch.njol.skript.expressions.base.SimplePropertyExpression;
+import ch.njol.skript.lang.Expression;
+import ch.njol.skript.lang.SkriptParser.ParseResult;
+import ch.njol.skript.util.Timespan;
+import ch.njol.util.Kleenean;
+import ch.njol.util.coll.CollectionUtils;
+
+@Name("Time Played")
+@Description("The amount of time a player has played for on the server.")
```
Comment (Blueyescat) [inline src/main/java/ch/njol/skript/expressions/ExprTimePlayed.java]: The description should include how it exactly works

File: src/main/java/ch/njol/skript/expressions/ExprTimePlayed.java (hunk 1 above)
Comment (ShaneBeee) [reply src/main/java/ch/njol/skript/expressions/ExprTimePlayed.java]: Good thinking!

File: src/main/java/ch/njol/skript/expressions/ExprTimePlayed.java (hunk 3 above)
Comment (ShaneBeee) [reply src/main/java/ch/njol/skript/expressions/ExprTimePlayed.java]: Oh right, I forgot to remove this. I added it for testing, as I was getting an error on one part.

File: src/main/java/ch/njol/skript/expressions/ExprTimePlayed.java [hunk 6]
```diff
@@ -0,0 +1,114 @@
...
+	@Override
+	public boolean init(Expression<?>[] exprs, int matchedPattern, Kleenean isDelayed, ParseResult parseResult) {
+		setExpr((Expression<Player>) exprs[0]);
+		return true;
+	}
+	
+	@Nullable
+	@Override
+	public Timespan convert(Player player) {
+		return Timespan.fromTicks_i((long) player.getStatistic(TIME_PLAYED) * 50);
```
Comment (Wealthyturtle) [inline src/main/java/ch/njol/skript/expressions/ExprTimePlayed.java]: Shouldn't multiply by 50 if you're constructing a Timespan from ticks

Comment (ShaneBeee) [reply src/main/java/ch/njol/skript/expressions/ExprTimePlayed.java]: oh crap, that's an oversight from when I changed it from `new TimeSpan(takes millis)`

File: src/main/java/ch/njol/skript/expressions/ExprTimePlayed.java [hunk 7]
```diff
@@ -0,0 +1,114 @@
...
+	"set player's time played to 0 seconds"})
+@Since("INSERT VERSION")
+public class ExprTimePlayed extends SimplePropertyExpression<Player, Timespan> {
+	
+	private final static Statistic TIME_PLAYED;
+
+	static {
+		register(ExprTimePlayed.class, Timespan.class, "time played", "players");
+		if (Skript.isRunningMinecraft(1, 13)) {
+			TIME_PLAYED = Statistic.PLAY_ONE_MINUTE; // Statistic name is misleading, it's actually measured in ticks
```
Comment (bensku) [inline src/main/java/ch/njol/skript/expressions/ExprTimePlayed.java]: This might change in future, so assuming this is dangerous.

Comment (ShaneBeee) [reply src/main/java/ch/njol/skript/expressions/ExprTimePlayed.java]: I have no clue why md_5 changed it from tick to minute, makes no sense. 
Anyways, theoretically anything could change in the future, I really hope this doesnt change again, but do you have a suggestion to make this better (not dangerous)?

Comment (bensku) [reply src/main/java/ch/njol/skript/expressions/ExprTimePlayed.java]: Unfortunately, no. Let's do this and just change it when/if Spigot breaks it.

**Pull request merged**